src/manifolds/Lorentz.jl

```diff
@@ -0,0 +1,40 @@
+@doc raw"""
+    LorentzMetric <: Metric
+
+Abstract type for Lorentz metrics, which have a single time dimension. These
+metrics assume the spacelike convention with the time dimension being last,
+giving the signature $(++...+-)$.
+"""
+abstract type LorentzMetric <: Metric end
+
+@doc raw"""
+    MinkowskiMetric <: LorentzMetric
+
+As a special metric of signature  $(++...+-)$, i.e. a [`LorentzMetric`](@ref)
+the minkowski metric on $\mathbb R^n$ is given by
+````math
+⟨a,b⟩_{\mathrm{M}} = -a_{n}b_{n} +
+\displaystyle\sum_{k=1}^{n-1} a_kb_k.
+````
+"""
+struct MinkowskiMetric <: LorentzMetric end
+
+@doc doc"""
+    Lorentz{N} = MetricManifold{Euclidean{N},LorentzMetric}
+
+The Lorentz manifold (or Lorentzian) is a pseudo-Riemannian manifold
+
+# Constructor
+
+    Lorentz(n[, metric=MinkowskiMetric()])
+
+Generate the Lorentz manifold of dimension `n` with the [`LorentzMetric`](@ref) `m`,
+which is by default set to the [`MinkowskiMetric`](@ref).
+"""
+const Lorentz = MetricManifold{Euclidean{Tuple{N,},ℝ},LorentzMetric} where N
+
+Lorentz(n,metric=MinkowskiMetric()) = Lorentz{n}(Euclidean(n),metric)
+
+local_metric(::Lorentz{N}, p) where {N} = Diagonal([ones(N-1)...,-1])
+
+inner(::Lorentz, p, X, Y) = -X[end] * Y[end] + sum(X[1:end-1] .* Y[1:end-1])
```

Comment:
Isn't this true only for the Minkowski metric and not all Lorentz metrics?

Reply:
Oh yes, you're right! Thanks for noticing. Line 38 also only holds for the `MinkowskiMetric` actually (since Lorentz is any with the same signature). I carefully read about that but must have missed it when looking for the float32 bug; there Line 38 introduced too many rounding errors and only line 40 was precise enough (and I think Lin e40 is also faster since there is no matrix multiplication necessary).